### PR TITLE
Added alignment and credit field to MediaCaptionTrait

### PR DESCRIPTION
**Pull request by balagan73**
MediaCaptionTrait now can display the credit field and can handle alignment.
Credits to @dipakmdhrm

### CODE REVIEW

File: web/modules/custom/server_general/src/MediaCaptionTrait.php [hunk 1]
```diff
@@ -21,19 +21,27 @@ trait MediaCaptionTrait {
...
+    $credit = $entity->hasField($credit_field_name) ? $this->getTextFieldValue($entity, $credit_field_name) : NULL;
+    if (empty($caption) && empty($credit)) {
+      // Nothing to output.
       return [];
     }
 
     return [
       '#theme' => 'server_theme_media_caption',
       '#caption' => $caption,
+      '#credit' => $credit,
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/MediaCaptionTrait.php:43]: I think we should change this to $items - that will get a render array. So we can pass caption, credit, etc'.

Then we could use the `verticalSpacing` -- we can pass alignment there.

**Review by amitaibu — CHANGES_REQUESTED** [review]
Thanks. Screenshots would be nice :)

In any case, I think we should change strategy, and move to composing the elements together

File: web/modules/custom/server_general/src/MediaCaptionTrait.php (hunk 1 above)
Comment (balagan73) [reply web/modules/custom/server_general/src/MediaCaptionTrait.php:43]: image

verticalSpacingTiny has gap-y-2 class, and here we have 8px gap. Should we create another vertical spacer with gap-y-1 class?

Comment (amitaibu) [reply web/modules/custom/server_general/src/MediaCaptionTrait.php:43]: Let's use `verticalSpacingTiny` as is

File: web/themes/custom/server_theme/server_theme.theme [hunk 2]
```diff
@@ -399,6 +399,7 @@ function server_theme_theme() {
   $info['server_theme_media_caption'] = [
     'variables' => [
       'caption' => NULL,
+      'credit' => NULL,
```
Comment (amitaibu) [inline web/themes/custom/server_theme/server_theme.theme:402]: => `$items => []`

**Comment by balagan73** [comment]
Then we can probably get rid of the templates for the caption, video and media.

**Comment by balagan73** [comment]
Style guide after the updates.

image

**Comment by balagan73** [comment]
Maybe deleting the video and image templates is not a good idea. Added wrappers for caption font colors.

File: web/themes/custom/server_theme/server_theme.theme (hunk 2 above)
Comment (amitaibu) [reply web/themes/custom/server_theme/server_theme.theme:402]: ^^. We should remove caption and credit, and call it `$items`

File: web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php [hunk 3]
```diff
@@ -285,17 +285,27 @@ protected function getCardsCentered(): array {
...
+    $elements[] = $caption;
+
+    // Credit.
+    $credit = $this->wrapTextResponsiveFontSize('Photo Credits: John Doe', 'base');
+    $credit = $this->wrapTextColor($credit, 'dark-gray');
+    $elements[] = $this->wrapTextItalic($credit);
+
+    $items = $this->wrapContainerVerticalSpacingTiny($elements, 'start');
 
     return [
```
Comment (amitaibu) [inline web/modules/custom/server_style_guide/src/Controller/StyleGuideController.php:305]: We shouldn't be calling these theme functions form StyleGuide - we should call the method on the trait. You can mock the $entity for that

**Comment by balagan73** [comment]
I will start again in a new branch, this went in a wrong way.

**Pull request closed without merging**